### ISSUE TITLE
RequiresUnsafeAttribute namespace handling diverges across four independent derivations

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#5721]
## Summary

Four components independently derive "does this member require unsafe?", and
they do not agree on which attribute namespace counts. An assembly using
`System.Runtime.CompilerServices.RequiresUnsafeAttribute` is reported as
requiring unsafe by two of them and not by the other two.

This is the same class of defect as #5670, found the same way: by inventorying
derivations while writing the Metadata semantic substrates design (#5273,
PR #5706).

## The four derivations

| Component | Namespaces accepted | Location |
| --- | --- | --- |
| Metadata `AttributeReader` | **both** `System.Diagnostics.CodeAnalysis` and `System.Runtime.CompilerServices` | `src/ILInspector.Metadata/AttributeReader.cs:297`-`306` |
| Analysis `HasRequiresUnsafe` | **both** | `src/ILInspector.Analysis/LibraryBodyPrimaryMetadataResolver.cs:265`-`270` |
| Decompiler `MethodDefinitionFacts` | **only** `System.Diagnostics.CodeAnalysis` | `src/ILInspector.Decompiler/Pipeline/MethodDefinitionFacts.cs:55`-`56` |
| CLI audit count (`RequiresUnsafeCount`) | **only** `System.Diagnostics.CodeAnalysis` | `src/ILInspector.Metadata/AssemblyDetailScanner.cs:147` |

Analysis is explicit that the divergence is deliberate on its side
(`LibraryBodyPrimaryMetadataResolver.cs:263`-`267`):

> Read attributes straight from SRM — a simple has-attribute check needs no
> shared decode/render machinery, so Analysis stays independent.
> Match the distinctive simple name: the implemented attribute is in
> System.Diagnostics.CodeAnalysis, while the design doc says
> System.Runtime.CompilerServices — tolerate the namespace churn.

That reasoning is sound in isolation. The problem is that three other
components made the same call independently and two reached the opposite
answer.

## Consequence

For a member carrying the `System.Runtime.CompilerServices` spelling:

- `dotnet-inspect` audit output undercounts `RequiresUnsafeCount`.
- The decompiler does not treat the method as requiring unsafe.
- Analysis and the metadata API surface do treat it as requiring unsafe.

The same assembly therefore gets contradictory answers depending on which
command the user runs.

## Relationship to the substrate work

`MemorySafetyMetadataIndex` already publishes a typed member unsafe contract
(`GetMemberContract`, `src/ILInspector.Metadata/MemorySafetyMetadataIndex.cs:363`)
that would settle this. **It has no caller in product code.** The typed answer
exists and is entirely unconsumed while four private raw decoders disagree.

PR #5706 records this family in the substrate inventory as satisfying the
duplication form of its admission test, with this issue as the tracked
divergence. Adoption is a separate effort per owner and is not proposed here.

## Suggested resolution

Decide the intended namespace set once, then have the consumers read the
substrate rather than each deciding again. Whether adoption happens per owner
or the shared check is simply widened is an owner decision; the invariant that
matters is that the four answers agree.

## Validation of this report

Each cited location was read at `main`. Reproduce with:

```bash
grep -n "RequiresUnsafeAttributeCompilerServices" src/ILInspector.Metadata/AttributeReader.cs
grep -n "RequiresUnsafeAttribute" src/ILInspector.Decompiler/Pipeline/MethodDefinitionFacts.cs
grep -n "RequiresUnsafeAttribute" src/ILInspector.Metadata/AssemblyDetailScanner.cs
sed -n '263,270p' src/ILInspector.Analysis/LibraryBodyPrimaryMetadataResolver.cs
grep -rn "GetMemberContract(" src --include=*.cs
```
